common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -400,10 +397,9 @@ private void publishLogEvent(TenantAwareLoggingEvent event)
                 arbitraryDataMap.put(columns[8], stacktrace);
             }
             if (payLoadData != null && arbitraryDataMap != null) {
-                Event laEvent = new Event("loganalyzer:1.0.0", System.currentTimeMillis(), null, null,
+                Event laEvent = new Event(streamDef, System.currentTimeMillis(), null, null,
                         payLoadData.toArray(), arbitraryDataMap);
                 dataPublisher.publish(laEvent);
```

Comment:
Yes, as per the above implementation when the exception occurred in the above try catch block publisher not get initialized properly. As a result, it gives the NPE logs when trying to publish the log events. 

Thank you for feedback regarding the issue when data publisher not properly initialize. We are fixed this as Appender tries to initialize data publisher for the first event checking nullity of the publisher object. If the publisher is not initialized log event processing will be stopped until user provide correct publisher configuration and restart the server. This will prevent multiple log messages.

Fixes for the new approach are in: 
https://github.com/wso2/shared-analytics/pull/131/commits/25411e1fa5d19d5db2ebc1b110a190d5267280b1

https://github.com/wso2/shared-analytics/commit/c46c5ba0ff6b2fb9e0dfb26e9896e7bd1ab99c9a